src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -742,27 +742,18 @@ private void ProcessMessageProc(object state)
         }
 
         private const string GUIDTAG = "PSGuid='";
-        private const int GUID_STR_LEN = 36;        // GUID string: 32 digits plus 4 dashes
 
-        private Guid GetMessageGuid(string data)
+        private bool ContainsMessageGuid(ReadOnlySpan<char> data)
         {
             // Perform quick scan for data packet for a GUID, ignoring any errors.
```

Comment:
Please add try/catch.  This method should not throw.

Reply:
With the new implementation the method can not throw. But I added explicit try-catch to avoid a regression if anybody change the method.

Reply:
Both Slice() and IndexOf() methods can throw.

Reply:
It seems ReadOnlySpan.IndexOf() never throw. Slice() could be but IndexOf() before it protects from this.
In any case I already added try-catch to exclude a regression in future..